#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.1198020675135694 0.9183371641167707 0.3772324450805386 3.8331219392672318
  position -1.3988231415003614 1.9741685289745712 -1.5892742575375465
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
  floorAppearance NULL
}
Robot {
  translation 0.800426 0.0298201 -0.7035
  rotation -0.0006411017888527957 0.9999996706495936 -0.0004976838360875723 -1.8325953071795862
  children [
    DEF IR_R DistanceSensor {
      translation -0.003 0 0.13
      rotation 0 0 -1 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "IR_R"
      type "infra-red"
    }
    DEF IR_R_ext DistanceSensor {
      translation -0.015 -1.59265e-05 0.13
      rotation 0 0 -1 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "IR_R_ext"
      type "infra-red"
    }
    DEF IR_L_ext DistanceSensor {
      translation 0.015 -1.59265e-05 0.13
      rotation 0 0 -1 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "IR_L_ext"
      type "infra-red"
    }
    DEF IR_L DistanceSensor {
      translation 0.003 0 0.13
      rotation 0 0 -1 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "IR_L"
      type "infra-red"
    }
    DEF OBJ_F DistanceSensor {
      translation -1.59265e-05 0.01 0.11
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "OBJ_F"
      numberOfRays 2
      aperture 0.3
    }
    DEF OBJ_B DistanceSensor {
      translation -0.0400477 0.01 -0.080064
      rotation 0 1 0 3.14
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "OBJ_B"
      aperture 3.14
    }
    DEF RodaFL HingeJoint {
      jointParameters HingeJointParameters {
        position 6149.932744456166
        anchor 0.04 0 0.04
        dampingConstant 5
      }
      device [
        RotationalMotor {
          name "MotorFL"
          maxVelocity 100
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.05000000000000002 0 0.04
        rotation 0.5308510479150851 -0.5300413530562428 -0.661251335709625 4.3123753043931075
        children [
          DEF Roda Shape {
            appearance PBRAppearance {
              baseColor 1 0.6 0.6
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        contactMaterial "RodaMaterial"
        boundingObject USE Roda
        physics DEF Physics_rpda Physics {
        }
      }
    }
    DEF RodaFR HingeJoint {
      jointParameters HingeJointParameters {
        position 8697.801244757082
        anchor -0.04 0 0.04
        dampingConstant 5
      }
      device [
        RotationalMotor {
          name "MotorFR"
          maxVelocity 100
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.05 0 0.04
        rotation -0.6341382999239109 0.633624509241092 -0.44315753165053734 3.978412510573371
        children [
          DEF Roda Shape {
            appearance PBRAppearance {
              baseColor 1 0.6 0.6
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        contactMaterial "RodaMaterial"
        boundingObject USE Roda
        physics DEF Physics_rpda Physics {
        }
      }
    }
    DEF RodaBL HingeJoint {
      jointParameters HingeJointParameters {
        position 6160.246612052427
        anchor 0.04 0 -0.05
        dampingConstant 5
      }
      device [
        RotationalMotor {
          name "MotorBL"
          maxVelocity 100
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.05000000000000002 0 -0.05
        rotation 0.6996242601463645 -0.6990673523125324 0.1477522640956744 2.8479803723515213
        children [
          DEF Roda Shape {
            appearance PBRAppearance {
              baseColor 1 0.6 0.6
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(2)"
        contactMaterial "RodaMaterial"
        boundingObject USE Roda
        physics DEF Physics_rpda Physics {
        }
      }
    }
    DEF RodaBR HingeJoint {
      jointParameters HingeJointParameters {
        position 6353.4132840757875
        anchor -0.04 0 -0.05
        dampingConstant 5
      }
      device [
        RotationalMotor {
          name "MotorBR"
          maxVelocity 100
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.049999999999999996 0 -0.05
        rotation -0.4673355233974411 0.46696351969655 -0.7506947314610977 4.430248587582048
        children [
          DEF Roda Shape {
            appearance PBRAppearance {
              baseColor 1 0.6 0.6
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(3)"
        contactMaterial "RodaMaterial"
        boundingObject USE Roda
        physics DEF Physics_rpda Physics {
        }
      }
    }
    Camera {
      translation 0.000201986 0.10781 -0.0995012
      rotation -0.00010359597428302096 0.9914447538807455 0.13052696759758944 3.14001
      fieldOfView 1
      width 1280
      height 720
    }
    DEF Base Shape {
      appearance PBRAppearance {
        baseColor 0.7 0.7 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.01 0.25
      }
    }
  ]
  boundingObject USE Base
  physics Physics {
  }
  controller "robot_controller"
}
Solid {
  translation -0.24 0.04 0.71
  rotation 0 -1 0 4.692820414042842e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry DEF Cubo_OBS Box {
        size 0.05 0.08 0.05
      }
    }
  ]
  name "solid(6)"
  boundingObject USE Cubo_OBS
}
Solid {
  translation -0.13 0.04 -0.794314
  rotation 0 1 0 -5.307179586466759e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry DEF Cubo_OBS Box {
        size 0.05 0.08 0.05
      }
    }
  ]
  name "solid(2)"
  boundingObject USE Cubo_OBS
}
Solid {
  translation 0.6 0.04 -0.184314
  rotation 0 -1 0 5.307156870259904e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry DEF Cilindro_OBS Cylinder {
        height 0.08
        radius 0.025
      }
    }
  ]
  name "solid(3)"
  boundingObject USE Cilindro_OBS
}
Solid {
  translation -0.59 0.04 -0.374314
  rotation 0 -1 0 5.307156870259904e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry DEF Cilindro_OBS Cylinder {
        height 0.08
        radius 0.025
      }
    }
  ]
  name "solid(4)"
  boundingObject USE Cilindro_OBS
}
Solid {
  translation -0.309998 0.04 -0.054314
  rotation 0 -1 0 5.307156870259904e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry DEF Cilindro_OBS Cylinder {
        height 0.08
        radius 0.025
      }
    }
  ]
  name "solid(5)"
  boundingObject USE Cilindro_OBS
}
Solid {
  translation 0 0.01 7.96327e-06
  rotation 1 0 0 -1.57
  scale 0.01 0.01 0.01
  children [
    Solid {
      name "obj_0"
    }
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.16862746 0.18039216 0.19215687
            roughness 1
            metalness 0
            name "color_2829873"
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                58.2 -71.8 0.029, 58.2 -71.8 0.044, -62.6 -71.8 0.044, 28 20.4 0.044, -2 20.4 0.029, -2 20.4 0.044, -2 19.4 0.044, 27 19.4 0.044, 27 -38.9 0.044, -1.1 -10 -0.014, -1.1 -10 -0.029, -1.1 20 -0.014, -1.1 20 -0.029, 28 20.4 0.029, 27 -38.9 0.029, 27 19.4 0.029, -32.1 50.1 0.029, -31.2 50.1 0.029, -31.2 50 0.029, -2.1 -10 -0.014, -2.1 20 -0.014, -2.1 -10 -0.029, -2.1 20 -0.029, -31.2 50.1 0.044, -32.1 50.1 0.044, -2 19.4 0.029, 57.2 50 0.029, -31.2 50 0.044, 57.2 50 0.044, -31.2 -38.9 0.029, -31.2 -38.9 0.044, 28 -39.9 0.044, -32.1 -39.9 0.044, -31.2 49.1 0.044, -61.6 78.7 0.044, -61.6 -70.8 0.044, -61.6 -70.8 0.029, -61.6 78.7 0.029, 58.2 50.1 0.044, 57.2 50.1 0.044, 57.2 49.1 0.044, 57.2 -70.8 0.044, -31.2 49.1 0.029, -32.1 -39.9 0.029, 28 -39.9 0.029, 58.2 50.1 0.029, 57.2 49.1 0.029, 57.2 50.1 0.029, 67.4 80.2 0.044, 67.4 80.2 0.029, -62.6 80.2 0.029, -62.6 80.2 0.044, 67.4 79.2 0.044, -62.6 79.2 0.044, 57.2 -70.8 0.029, 67.4 79.2 0.029, -62.6 79.2 0.029, -62.6 78.7 0.044, -62.6 78.7 0.029, -62.6 -71.8 0.029
              ]
            }
            coordIndex [
              0, 1, 2, -1, 3, 4, 5, -1, 5, 6, 7, -1, 5, 7, 3, -1, 8, 3, 7, -1, 9, 10, 11, -1, 12, 11, 10, -1, 13, 14, 15, -1, 16, 17, 18, -1, 13, 4, 3, -1, 19, 20, 21, -1, 22, 21, 20, -1, 23, 17, 16, -1, 23, 16, 24, -1, 11, 20, 9, -1, 10, 22, 12, -1, 11, 12, 22, -1, 11, 22, 20, -1, 13, 15, 4, -1, 25, 4, 15, -1, 26, 18, 27, -1, 26, 27, 28, -1, 9, 20, 19, -1, 14, 29, 30, -1, 14, 30, 8, -1, 31, 8, 32, -1, 33, 32, 30, -1, 34, 35, 36, -1, 34, 36, 37, -1, 38, 39, 28, -1, 38, 28, 40, -1, 1, 38, 41, -1, 22, 10, 21, -1, 3, 8, 31, -1, 8, 30, 32, -1, 42, 29, 16, -1, 16, 29, 43, -1, 29, 14, 43, -1, 19, 21, 10, -1, 19, 10, 9, -1, 27, 24, 33, -1, 28, 27, 40, -1, 33, 40, 27, -1, 32, 43, 44, -1, 32, 44, 31, -1, 5, 4, 25, -1, 5, 25, 6, -1, 44, 43, 14, -1, 13, 44, 14, -1, 45, 0, 46, -1, 45, 46, 26, -1, 47, 45, 26, -1, 7, 6, 25, -1, 7, 25, 15, -1, 13, 3, 44, -1, 31, 44, 3, -1, 15, 14, 8, -1, 15, 8, 7, -1, 42, 46, 40, -1, 38, 45, 47, -1, 38, 47, 39, -1, 33, 30, 29, -1, 33, 29, 42, -1, 40, 33, 42, -1, 26, 46, 42, -1, 16, 18, 42, -1, 26, 42, 18, -1, 48, 49, 50, -1, 48, 50, 51, -1, 16, 43, 32, -1, 16, 32, 24, -1, 17, 23, 27, -1, 17, 27, 18, -1, 40, 46, 41, -1, 48, 51, 52, -1, 53, 52, 51, -1, 32, 33, 24, -1, 23, 24, 27, -1, 35, 41, 36, -1, 54, 36, 41, -1, 54, 41, 46, -1, 49, 55, 56, -1, 49, 56, 50, -1, 51, 50, 56, -1, 51, 56, 53, -1, 39, 47, 26, -1, 39, 26, 28, -1, 40, 41, 38, -1, 52, 55, 49, -1, 52, 49, 48, -1, 57, 58, 59, -1, 57, 59, 2, -1, 41, 35, 1, -1, 2, 35, 57, -1, 1, 35, 2, -1, 35, 34, 57, -1, 58, 36, 59, -1, 37, 36, 58, -1, 2, 59, 0, -1, 0, 59, 36, -1, 36, 54, 0, -1, 46, 0, 54, -1, 34, 37, 58, -1, 34, 58, 57, -1, 52, 53, 56, -1, 52, 56, 55, -1, 38, 1, 0, -1, 38, 0, 45, -1
            ]
          }
        }
      ]
      name "group_0_2829873"
    }
  ]
  name "tinker.obj(1)"
}
